calico_rkt/calico_rkt.py

```diff
@@ -1,198 +1,212 @@
-    ip = _allocate_IP()
+    endpoint, ip = _container_add(hostname=HOSTNAME,
+                                  orchestrator_id=ORCHESTRATOR_ID,
+                                  container_id=container_id,
+                                  netns_path=netns_path,
+                                  interface=interface,
+                                  subnet=subnet)
+
+    _log.info('Finished configuring network interface')
+    return endpoint, ip
+
+
+def _container_add(hostname, orchestrator_id, container_id, netns_path, interface, subnet):
+    """
+    Add a container to Calico networking
+    Return Endpoint object and newly allocated IP
+    """
+    # Allocate and Assign ip address through IPAM _datastore_client
+    pool = _generate_pool(subnet)
+    ip = _allocate_ip(pool)
 
     # Create Endpoint object
     try:
-        ep = client.create_endpoint(HOSTNAME, ORCHESTRATOR_ID,
-                                      container_id, [IPAddress(ip)])
+        ep = _datastore_client.create_endpoint(HOSTNAME, ORCHESTRATOR_ID,
+                                               container_id, [ip])
     except AddrFormatError:
-        print_stderr("This node is not configured for IPv%d. Unassigning IP "\
-                      "address %s then exiting."  % ip.version, ip)
-        client.unassign_address(IPNetwork(input_json['ipam']['subnet']), ip)
+        _log.error("This node is not configured for IPv%d. Unassigning IP "
+                   "address %s then exiting." % ip.version, ip)
+        _datastore_client.unassign_address(pool, ip)
         sys.exit(1)
 
     # Create the veth, move into the container namespace, add the IP and
     # set up the default routes.
     ep.mac = ep.provision_veth(Namespace(netns_path), interface)
-    client.set_endpoint(ep)
+    _datastore_client.set_endpoint(ep)
 
-    # Let the caller know what endpoint was created.
     return ep, ip
 
-def _container_remove(hostname, orchestrator_id, container_id, client):
+
+def _container_remove(hostname, orchestrator_id, container_id):
     """
     Remove the indicated container on this host from Calico networking
     """
     # Find the endpoint ID. We need this to find any ACL rules
     try:
-        endpoint = client.get_endpoint(
-            hostname=hostname,
-            orchestrator_id=orchestrator_id,
-            workload_id=container_id
-        )
+        endpoint = _datastore_client.get_endpoint(hostname=hostname,
+                                                  orchestrator_id=orchestrator_id,
+                                                  workload_id=container_id)
     except KeyError:
-        print_stderr("Container %s doesn't contain any endpoints" % container_id)
+        _log.error(
+            "Container %s doesn't contain any endpoints" % container_id)
         sys.exit(1)
 
     # Remove any IP address assignments that this endpoint has
     for net in endpoint.ipv4_nets | endpoint.ipv6_nets:
         assert(net.size == 1)
```

Comment:
Add a message here, otherwise if this assertion is ever hit the logs will just read `Assertion Error` and with no stack trace or message its pretty frustrating to figure out which one was hit: `assert net.size == 1, "assertion message yadda"`